src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -1133,14 +1133,29 @@ void QgsLayoutItemLegend::doUpdateFilterByMap()
 
     mapSettings.setExpressionContext( createExpressionContext() );
 
-    const QgsGeometry atlasGeometry = mInAtlas ? mLayout->reportContext().currentGeometry( mapSettings.destinationCrs() ) : QgsGeometry();
+    const QgsGeometry atlasGeometry { mLayout->reportContext().currentGeometry( mapSettings.destinationCrs() ) };
 
     QgsLayerTreeFilterSettings filterSettings( mapSettings );
 
+    QList<QgsMapLayer *> layersToClip;
+    if ( !atlasGeometry.isNull() && mMap->atlasClippingSettings()->enabled() )
+    {
+      layersToClip = mMap->atlasClippingSettings()->layersToClip();
+      for ( QgsMapLayer *layer : std::as_const( layersToClip ) )
```

Comment:
```suggestion
      const QList<QgsMapLayer *>  layersToClip = mMap->atlasClippingSettings()->layersToClip();
      for ( QgsMapLayer *layer : layersToClip) 
```

We can reduce the scope of layersToClip here

Reply:
Yes, the `std::as_const` should stay though even if it is redundant: I was told several times from other developers that it makes things clearer.

Reply:
Question: isn't my original version slightly faster? I don't know if the compiler can optimize it.

Reply:
Yes, we can keep the std::as_const indeed, it doesn't harm.

Not sure to understand why it would be slighly faster. Anyway it is purely cosmetic and non blocking for merge

Reply:
> Yes, we can keep the std::as_const indeed, it doesn't harm.
> 
> Not sure to understand why it would be slighly faster. Anyway it is purely cosmetic and non blocking for merge

because the assignment down inside the loop is done multiple times instead of once before the loop.

Reply:
my comment was just to move the declaration at the same line of the assignment, not inside the loop. But nevermind that's not a big deal.